### PR TITLE
runHost function

### PR DESCRIPTION
**Pull request by ElvishJerricco**
Adding a function for creating new hosts with PerformEvent and PostBuild easily. Currently having some trouble getting it to work in the context we need, and haven't documented it yet, so WIP

### CODE REVIEW

File: src/Reflex/Main.hs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import Data.Maybe (catMaybes)
+import Control.Monad.IO.Class
+import Control.Monad (forever)
+import Data.Traversable (for)
+import Data.Foldable (for_)
+import Data.Functor.Identity
+
+type EventChannel t = Chan [DSum (EventTriggerRef t) TriggerInvocation]
+
+runHost
```
Comment (ryantrinkle) [inline src/Reflex/Main.hs]: Let's give this a slightly longer name, like `runHostWithIO`, to indicate that it grants access to IO.  I don't want people to get the impression that all hosts can/should support IO.

**Comment by dalaing** [comment]
If you wanted to take anything from https://github.com/qfpl/reflex-basic-host/blob/master/src/Reflex/Host/Basic.hs for this, feel free.

It started life as almost exactly what you have here, then I needed the ability to sometimes use an Event to shutdown the host, which is where it started to diverge, and then I needed a newtype and the various instances to use this in a context where the pile-o'-constraints approach was giving me grief.

**Comment by ryantrinkle** [comment]
Having an event to shut down reflex-dom apps would be great :)

**Comment by dalaing** [comment]
I've been looking at adding that kind of event to as many of the reflex-dom backends as I can, for use with criterion and QuickCheck/Hedgehog...

**Comment by ryantrinkle** [comment]
@dalaing It sounds like it would be great to base reflex-dom on that!

**Comment by ali-abrar** [comment]
It would be nice if something like this also worked with reflex-vty.

**Comment by ali-abrar** [comment]
https://github.com/reflex-frp/reflex-vty/blob/develop/src/Reflex/Vty/Host.hs#L89

**Comment by Ericson2314** [comment]
OK Good CI works. Next step is to make reflex-vty use this.

**Comment by endgame** [comment]
Is this still relevant given that `Reflex.Host.Headless` has been released?

**Comment by ali-abrar** [comment]
I think the point of this was to provide something to make it easier to implement headless, vty, or other hosts.

**Pull request closed without merging**